### PR TITLE
SITES-8730: add integration tests to cover teaser corner cases

### PR DESCRIPTION
**Pull request by Buuhuu**
This adds an integration test that verifies the teaser markup. In particular it tests the teaser v1 and teaser v2 markup with a CTA action referencing an Asset.

### CODE REVIEW

File: testing/it/http/src/test/java/com/adobe/cq/wcm/core/components/it/http/ComponentsIT.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
```
Comment (jckautzmann) [inline testing/it/http/src/test/java/com/adobe/cq/wcm/core/components/it/http/ComponentsIT.java:2]: Should be 2022

**Comment by jckautzmann** [comment]
LGTM, thx! I would also add tests to cover CTAs linking to assets in Teaser v1 and v2, as theses use cases also trigger the Class Cast Exception.

**Review by jckautzmann — APPROVED** [review]
